### PR TITLE
CMDCT-4086: Add / Edit Report Modal (NAAAR)

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Finalizing the add / edit report modal for NAAAR, including the frontend and a couple of changes needed for the backend.

### Related ticket(s)

CMDCT-4086

---
### How to test

- Log in as a state user
- Create a NAAAR report
- Click on the "Edit" icon to verify that the fields are hydrating correctly as well

You should see the state name populating upon entering the first page of the report.

If you run `DYNAMO_ENDPOINT=http://localhost:8000 dynamodb-admin` in a new terminal and visit `localhost:8001` you will see the report metadata as an entry.

**NOTE:** The dashboard table is not populating correctly, there is a separate ticket to address this [here](https://jiraent.cms.gov/browse/CMDCT-4116)

### Notes

N/A

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by jessabean — APPROVED** [review]
👍🏻  Works as expected! Just a non-blocking nit and a question.

**Review by gmrabian — COMMENTED** [review]
the info populated in the dashboard table does not line up with the column names
i tested with both a custom and a provided plan type

Screenshot 2024-11-14 at 10 34 15 AM

File: services/ui-src/src/components/modals/AddEditReportModal.tsx [hunk 1]
```diff
@@ -227,6 +243,7 @@ export const AddEditReportModal = ({
       content={{
         heading: selectedReport?.id ? form.heading?.edit : form.heading?.add,
         subheading: selectedReport?.id ? "" : form.heading?.subheading,
+        hint: selectedReport?.id ? "" : form.heading?.hint,
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/ui-src/src/components/modals/AddEditReportModal.tsx]: That's a good point, I've just updated it to `intro`!

File: services/ui-src/src/forms/addEditNaaarReport/addEditNaaarReport.json [hunk 2]
```diff
@@ -4,17 +4,91 @@
...
+      }
+    },
+    {
+      "id": "reportingPeriodStartDate",
+      "type": "date",
+      "validation": "date",
+      "props": {
+        "label": "Reporting period (i.e. contract period) start date",
+        "hint": "Enter start date of the reporting period represented in the report.",
+        "timetype": "startDate"
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/ui-src/src/forms/addEditNaaarReport/addEditNaaarReport.json:36]: I just checked, looks like it is meant for validation of the time here: https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/5c058821d4a0fc55f29b1800199b299dc1452409/services/ui-src/src/utils/other/time.ts#L122 but I'm not sure if we actually need it...

**Comment by karla-vm** [comment]
> the info populated in the dashboard table does not line up with the column names i tested with both a custom and a provided plan type
> 
> Screenshot 2024-11-14 at 10 34 15 AM

Good catch! I should add this to the description, but there is a separate ticket for the table: https://jiraent.cms.gov/browse/CMDCT-4116

**Review by jessabean — APPROVED** [review]

**Review by gmrabian — APPROVED** [review]

**Review by bangbay-bluetiger — APPROVED** [review]

**Pull request merged**